### DISCUSSION TITLE
Register-PnPAzureADApp giving an error - Cannot validate argument on parameter 'SharePointApplicationPermissions'

### DISCUSSION THREAD

**Discussion by PratikD21** [pnp/powershell#1468] [Q&A]
While executing below cmdlet getting error as per below image. But, from Azure App Registration Portal, "Sites.Selected" is present for selection in API Permission.

Register-PnPAzureADApp -ApplicationName SPAutomation -Tenant zzzz.onmicrosoft.com -SharePointApplicationPermissions "Sites.Selected" -OutPath C:\Certificate\ -CertificatePassword (ConvertTo-SecureString -String "password" -AsPlainText -Force) -CommonName MyCertificate -DeviceLogin

Error : 
![image](https://user-images.githubusercontent.com/40222054/147094884-f9418909-428b-4d6c-8b31-3dc00d9a8f37.png)

**Comment by LeonArmston** [comment]
@PratikD21 - Not sure if you resolved this but I did it this way for Sites.Selected

```powershell
Register-PnPAzureADApp -ApplicationName SitesResourceSpecific -Tenant contoso.onmicrosoft.com -Store CurrentUser -GraphApplicationPermissions "Sites.Selected" -Username "user@contoso.com" -Password (Read-Host -AsSecureString -Prompt "Enter Password")
```

**Reply by PratikD21** [reply]
Hi @LeonArmston,

Thanks for response. I have tried with command which you provided and facing and issue where permission are not getting set for app after app get created with mentioned command. I am getting an error as mentioned below links. Could you please check and help me with this.

•	https://github.com/pnp/powershell/issues/1270
•	https://github.com/pnp/powershell/issues/917

Regards,
Pratik
